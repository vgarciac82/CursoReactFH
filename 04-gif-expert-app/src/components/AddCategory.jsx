import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory  }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChanged = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length <= 1) return;

  
    onNewCategory(inputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar..."
        value={inputValue}
        onChange={onInputChanged}
      />
    </form>
  );
};


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}