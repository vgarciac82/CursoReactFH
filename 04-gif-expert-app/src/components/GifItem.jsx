<<<<<<< HEAD
import PropTypes from 'prop-types'

export const GifItem = ({ title, url, id }) => {
=======
import React from "react";
import PropTypes from "prop-types";
>>>>>>> 00db8db4b2d38759da756ea7a85e019058b99618

  return (
    <div className="card">
        <img src={ url } alt={ title } />
        <p>{ title }</p>
    </div>
<<<<<<< HEAD
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired, 
  url  : PropTypes.string.isRequired, 
}
=======
  );
};


GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
>>>>>>> 00db8db4b2d38759da756ea7a85e019058b99618
