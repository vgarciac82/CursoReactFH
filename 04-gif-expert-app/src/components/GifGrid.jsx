import React, { useState } from "react";
import { useEffect } from "react";
import getGifs from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {
        isLoading && (<p>Loading...</p>)
      }
      <div className="card-grid">
        {images.map((image) => {
          return <GifItem key={image.id} {...image}></GifItem>;
        })}
      </div>
    </>
  );
};
