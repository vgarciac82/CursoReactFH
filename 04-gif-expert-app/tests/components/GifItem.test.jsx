import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Prueba en <GifItem/>", () => {
  const title = "Saitama";
  const url = "https://media.giphy.com/media/3o7TKsQ8gqVrXwXWqM/giphy.gif";

  test("Debe mostrar match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe mostrar la imagen con el IURL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);

    // expect( screen.getByRole("img").src ).toBe(url);
    // expect( screen.getByRole("img").alt ).toBe(title);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("debe mostrar el titulo", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
