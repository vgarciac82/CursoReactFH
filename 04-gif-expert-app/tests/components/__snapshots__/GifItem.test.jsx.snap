// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifItem/> Debe mostrar match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://media.giphy.com/media/3o7TKsQ8gqVrXwXWqM/giphy.gif"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
